### ISSUE TITLE
ActionView is not updated

### ISSUE THREAD

**Issue by ai212983** [JakeWharton/ActionBarSherlock#99]
Sorry for adding lame issue question (I have less than one week experience with android), but there's noone else to ask.

I'm inflating ActionBar's menu with following item:

``` java
    <item android:id="@+id/menu_checkout"
        android:title="Checkout"
        android:showAsAction="ifRoom"
        android:actionLayout="@layout/action_item_checkout" />
```

action_item_checkout is supposed to be an action view:

``` java
<?xml version="1.0" encoding="utf-8"?>

<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
    android:layout_width="wrap_content"
    android:layout_height="match_parent"
    android:orientation="horizontal" android:padding="5dp">

<ImageView android:id="@+id/checkout_icon"
    android:src="@drawable/ic_menu_shopping_holo_light"
    android:paddingRight="10dip"
    android:layout_width="wrap_content"
    android:layout_height="match_parent" />

<TextView android:id="@+id/checkout_text"
    android:text="2 125"
    android:textAppearance="@style/TextAppearance.ActionBar.Sum"
    android:singleLine="true"
    android:gravity="fill"
    android:layout_height="match_parent"
    android:layout_width="wrap_content" />

</LinearLayout>
```

In my code in onCreateOptionsMenu method I'm getting reference to this text view and changing it: 

``` java
        MenuItem checkout = menu.findItem(R.id.menu_checkout);
        mCheckout = checkout;
        mCheckoutTextView = (TextView) checkout.getActionView().findViewById(R.id.checkout_text);
        mCheckoutTextView.setText("wooot");  // XXX: Somewhy it changes nothing

        Log.w("MyApp", "menuitem: " + checkout + ", textView: " + mCheckoutTextView + ", its text value: " + mCheckoutTextView.getText() );
```

Log tells me its text has changed, but nothing changes in ActionBar itself. Is it a bug or I'm doing something wrong here?

**Comment by ai212983** [comment]
That's funny. If in onCreateOptions method we will remove ActionView and reattach it again, then everything will work:

``` java
@Override
public boolean onCreateOptionsMenu(Menu menu) {
    getMenuInflater().inflate(R.menu.main_menu, menu);

    MenuItem checkout = menu.findItem(R.id.menu_checkout);
    View checkoutActionView = checkout.getActionView();
    mCheckoutTextView = (TextView) checkoutActionView.findViewById(R.id.checkout_text);

    checkout.setActionView(null);
    checkout.setActionView(checkoutActionView);

    mCheckoutTextView.setText("wooot");  // XXX: Now it will work fine! -df

    Log.w("Bento", "menuitem: " + checkout + ", textView: " + mCheckoutTextView + ", its text value: " + mCheckoutTextView.getText() );

    return super.onCreateOptionsMenu(menu);
}
```

**Comment by JakeWharton** [comment]
Does this happen on both pre-3.0 and post-3.0 devices? Or is it only happening on pre-3.0?

**Comment by JakeWharton** [comment]
FYI the forthcoming version 3.5 (currently in the `dev` branch) should fix this as it uses the exact same classes as Android 4.0 for menu and action item control.

**Comment by ai212983** [comment]
Thanks. I'm totally overloaded here (sry, has no time to check this bug on post-3 devices),  but I'll check it out once 3.5 is out. Any estimations on release?

And thanks for your totally awesome work, btw. =)

**Comment by JakeWharton** [comment]
Couple days max. As of tonight it's stable enough for testing just going to put the word out to some people to try it in their apps for a final testing.

**Comment by JakeWharton** [comment]
Released with 3.5.0.

**Comment by bhanupriyach** [comment]
still not working
